testsuite/tests/typing-short-paths/short-paths.compilers.reference

```diff
@@ -96,6 +96,7 @@ Error: Signature mismatch:
          type t = int
        is not included in
          type t = string
+       The type t is not equal to the type string
```

Comment:
I think it would be very difficult to change this behaviour. We need to print the type in an environment that includes `t`, so short-paths is always going to have the option of choosing it.

Reply:
A mention in the test that there is a potential non-optimal choice would be fine with me.

Reply:
Naive question: printtyp has a mechanism for "hiding" some names for similar (internal) purposes. Could it be made available to clients to be used here? (I would guess the answer is "not easily", otherwise @Octachron would have suggested it.)

Reply:
Part of the issue is that this "hiding" should be contextual: it depends on the previous elements of the trace, and it should be only applied when printing. That sounds complex (and unrelated) enough that is better to left this refinement to another day's investigation.